plugins/inputs/vsphere/endpoint.go

```diff
@@ -1204,7 +1204,8 @@ func (e *Endpoint) collectChunk(
 			continue
 		}
 		buckets := make(map[string]metricEntry)
-		for _, v := range em.Value {
+		for i := range em.Value {
+			v := em.Value[i]
```

Comment:
I think here the better alternative is to change `populateTags` to not take a `performance.MetricSeries` pointer but a value
```diff
- func (e *Endpoint) populateTags(objectRef *objectRef, resourceType string, resource *resourceKind, t map[string]string, v *performance.MetricSeries) {
+ func (e *Endpoint) populateTags(objectRef *objectRef, resourceType string, resource *resourceKind, t map[string]string, v performance.MetricSeries) {
```

The value of `v` is not changed in that function so the for loop can stay as-is.